libnavui-status/src/main/java/com/mapbox/navigation/ui/status/model/Status.kt

```diff
@@ -0,0 +1,41 @@
+package com.mapbox.navigation.ui.status.model
+
+import androidx.annotation.DrawableRes
+
+/**
+ * This class stores information to be displayed by the [MapboxStatusView].
+ */
+data class Status(
```

Comment:
@tomaszrybakiewicz @abhishek1508 having a public constructor like this will be a major semver if we ever want to add a field to it. For this reason we've been using Builder, Mapper, Processor classes

Reply:
Event with default values adding a new field will be a breaking change, because of the generated `copy` method, which doesn't have default values for Java users.

Reply:
As long as we provide default values for the new fields we should be fine.
For the Java users we can add backward compatible `copy` method.
eg.
```kotlin
data class Status(
 ...
 val newField: Boolean = false
) {
    fun copy(
        message: String = this.message,
        duration: Long = this.duration,
        animated: Boolean = this.animated,
        spinner: Boolean = this.spinner,
        icon: Int = this.icon
    ) = Status(message, duration, animated, spinner, icon, this.newField)
}
```

Reply:
You also need to add `@JvmOverloads` annotation to the constructor. 
I would still prefer to see this class reimplemented using Builder pattern for consistency with the rest of SDK. We have `BuilderTest` that will check your implementation. Adding a new copy method with each new field doesn't look appealing to me.

Reply:
I think on the UI side of the SDK, we have been using `Builder` only when a UI Component has multiple configuration options. There are 2 different kinds of model classes we have been following on the UI side:
- models which cannot be created by an end user: For eg: `JunctionValue`
- models which can be created by an end user: For eg: `Maneuver`
Both these kind of model classes are simple class with an internal keyword: `class Maneuver internal constructor(...)`, however for the models of 2nd type which an end user can create, we associate a Factory with it: `ManeuverFactory` with an `Experimental` annotation. The purpose of the `Experimental` annotation is to inform the user that there might be in future instances where we can add/remove new properties. Even though, I am not a big fan of it and there is a potential to improve, I would suggest to use the same pattern here to keep the consistency with other UI components.